concordium-consensus/src/Concordium/KonsensusV1/Consensus/Blocks.hs

```diff
@@ -1229,6 +1267,53 @@ bakeBlock BakeBlockInputs{..} = do
     pendingTransactionTable .=! newPTT
     return signedBlock
 
+-- | Extract information from SkovData and the block state to compute the result block hash.
+computeBlockResultHash ::
+    forall m.
+    ( MonadState (SkovData (MPV m)) m,
+      BlockStateStorage m,
+      BlockState m ~ HashedPersistentBlockState (MPV m),
+      IsConsensusV1 (MPV m)
+    ) =>
+    -- | The block state right after executing the block.
+    HashedPersistentBlockState (MPV m) ->
+    -- | The relative block height for the block.
+    BlockHeight ->
+    m BlockResultHash
+computeBlockResultHash newState relativeBlockHeight = do
+    theBlockStateHash <- getStateHash newState
+    transactionOutcomesHash <- getTransactionOutcomesHash newState
+    currentFinalizationCommitteeHash <- use $ to bakersForCurrentEpoch . bfFinalizerHash
+    nextFinalizationCommitteeHash <- do
+        -- Attempt to get the finalization committee from SkovData otherwise compute it from the
+        -- information in the block state.
+        currentEpoch <- use (roundStatus . rsCurrentEpoch)
+        nextSkovBakersAndFinalizers <- gets (getBakersForEpoch (currentEpoch + 1))
+        case nextSkovBakersAndFinalizers of
+            Just bakersAndFinalizers -> return $ bakersAndFinalizers ^. bfFinalizerHash
+            Nothing -> do
+                nextFullBakers <- getNextEpochBakers newState
+                nextFinalizationParameters <- getNextEpochFinalizationCommitteeParameters newState
+                let nextFinalizationCommittee = computeFinalizationCommittee nextFullBakers nextFinalizationParameters
+                return $ computeFinalizationCommitteeHash nextFinalizationCommittee
```

Comment:
This relies on the epoch of the block matching the current epoch (as in `roundStatus . rsCurrentEpoch`). This should be documented as a precondition of the function, and we should ensure that it holds true everywhere it's used.

Reply:
Actually, I'm not entirely sure that this is guaranteed to be the case in `processBlock`. Instead, I would suggest passing in the epoch of the block and then using `getBakersForEpoch` on that, falling back to computing them from the block state.

Reply:
I have changed this and also remove the use of `bakersForCurrentEpoch` above this.